core/src/main/java/org/apache/iceberg/io/PartitionedWriter.java

```diff
@@ -63,10 +66,29 @@ public void write(InternalRow row) throws IOException {
         throw new IllegalStateException("Already closed files for partition: " + key.toPath());
       }
 
-      setCurrentKey(key.copy());
-      openCurrent();
+      currentKey = key.copy();
+    }
+
+    if (currentAppender == null) {
+      currentAppender = new RollingFileAppender(currentKey);
```

Comment:
It would be nice to not change the logic for opening an appender. Before, this was part of the flow of changing partitions.

Reply:
Now we've changed to maintain the partitionKey inside the `RollingFileWriter` (we've discussed this before , this is because for fanout writer we may have multiple writers append records),  so the  RollingFileAppender creation is actually doing the partition key setting.  I did not open the appender here because we only need to open an appender when there's a real record to write (in case of opening an appender without writing a record) ,  all those logic have been hidden inside the RollingFileAppender.